be/src/olap/comparison_predicate.h

```diff
@@ -17,46 +17,475 @@
+                                                           _value, _is_greater(), _is_eq())
+                                                 : dict_column_ptr->find_code(_value);
+                    auto& data_array = dict_column_ptr->get_data();
+
+                    _base_loop<true>(size, flags, &null_map, data_array, dict_code);
+                } else {
+                    LOG(FATAL) << "column_dictionary must use StringValue predicate.";
+                }
+            } else {
+                auto& data_array = reinterpret_cast<const vectorized::PredicateColumnType<TReal>&>(
+                                           nested_column)
+                                           .get_data();
+
+                _base_loop<true>(size, flags, &null_map, data_array, value_real);
+                for (uint16_t i = 0; i < size; i++) {
+                }
+            }
+        } else {
+            if (column.is_column_dictionary()) {
+                if constexpr (std::is_same_v<T, StringValue>) {
+                    auto* dict_column_ptr =
+                            vectorized::check_and_get_column<vectorized::ColumnDictI32>(column);
+                    auto dict_code = _is_range() ? dict_column_ptr->find_code_by_bound(
+                                                           _value, _is_greater(), _is_eq())
+                                                 : dict_column_ptr->find_code(_value);
+                    auto& data_array = dict_column_ptr->get_data();
+
+                    _base_loop<false>(size, flags, nullptr, data_array, dict_code);
+                } else {
+                    LOG(FATAL) << "column_dictionary must use StringValue predicate.";
+                }
+            } else {
+                auto& data_array =
+                        vectorized::check_and_get_column<vectorized::PredicateColumnType<TReal>>(
+                                column)
+                                ->get_data();
+
+                _base_loop<false>(size, flags, nullptr, data_array, value_real);
+            }
+        }
+
+        if (_opposite) {
+            for (uint16_t i = 0; i < size; i++) {
+                flags[i] = !flags[i];
+            }
+        }
+    }
+
+private:
+    template <typename LeftT, typename RightT>
+    bool _operator(const LeftT& lhs, const RightT& rhs) const {
+        if constexpr (PT == PredicateType::EQ) {
+            return lhs == rhs;
+        } else if constexpr (PT == PredicateType::NE) {
+            return lhs != rhs;
+        } else if constexpr (PT == PredicateType::LT) {
+            return lhs < rhs;
+        } else if constexpr (PT == PredicateType::LE) {
+            return lhs <= rhs;
+        } else if constexpr (PT == PredicateType::GT) {
+            return lhs > rhs;
+        } else if constexpr (PT == PredicateType::GE) {
+            return lhs >= rhs;
+        }
+    }
+
+    constexpr bool _is_range() const {
```

Comment:
Not very accurate.     PredicateType could be
    EQ = 1,
    NE = 2,
    LT = 3,
    LE = 4,
    GT = 5,
    GE = 6,
    IN_LIST = 7,
    NOT_IN_LIST = 8,
    IS_NULL = 9,
    IS_NOT_NULL = 10,
    BF = 11, // BloomFilter

better use > < >= <=